templates/default/elk-nginx-proxy-conf.erb

```diff
@@ -0,0 +1,59 @@
+# This file is installed automatically by chef. Do not edit or delete.
+
+server {
+  listen 80;
+  return 301 https://$host$request_uri;
+}
+
+server {
+  listen 443 ssl;
+  listen [::]:443 ipv6only=on ssl;
+
+  ssl_certificate     ssl/certificate.cert;
+  ssl_certificate_key ssl/certificate.key;
+
+  server_name kibana.interface;
+
+  root /usr/share/nginx/html;
+  index index.html index.htm;
+
+  # Some settings are recommended by https://raymii.org/s/tutorials/Strong_SSL_Security_On_nginx.html
+  # Need SSLv3 for IE on some older windows, this is the default set.
+  # ssl_protocols SSLv3 TLSv1 TLSv1.1 TLSv1.2;
+  ssl_prefer_server_ciphers on;
+  ssl_session_cache shared:SSL:10m;
+
+  auth_basic "Restricted";
+  auth_basic_user_file /etc/nginx/conf.d/kibana.htpasswd;
+  
+  proxy_buffering off;
+  proxy_http_version 1.1;
+  proxy_read_timeout 30m;
+  proxy_send_timeout 30m;
+  proxy_set_header Host $host;
+  proxy_set_header X-Real-IP $remote_addr;
+  proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+  proxy_set_header Upgrade $http_upgrade;
+  proxy_set_header Connection 'upgrade';
+  proxy_set_header Host $host;
+  proxy_set_header X-Forwarded-SSL on;
+  proxy_cache_bypass $http_upgrade;
+
+  client_max_body_size 102400m;
+  gzip on;
+
+  location / {
+    proxy_pass http://localhost:5601;
+  }
+  
+  location ~ ^/es.*$ {
+    proxy_pass http://<%= @elasticsearch_host %>:9200;
```

Comment:
the `auth_basic` settings defined above this cover all of these locations. There are more sophisticated access/auth mechanisms I plan to explore (e.g., [Elasticsearch Shield](https://www.elastic.co/products/shield)), but for now I'm hoping basic auth + forcing https is enough.